### ISSUE TITLE
Unable to load checkpoints for Nasnet?

### ISSUE THREAD

**Issue by alyahmed** [tensorflow/models#3938]
Seems like the latest checkpoint file is not compatible with the latest repository commit + Tensorflow version? The model trains just fine but clearly the checkpoint has a naming mismatch with what is expected during training. The model will train but never converge, the loss function will just fluctuate.

I'm currently using:
Commit -> 6741cfce9deacdd1f930fd70f9e4550461f5888d

`Merge: 18e0643 13c9de3
Author: aquariusjay <aquariusjay@users.noreply.github.com>
Date:   Mon Apr 9 10:37:52 2018 -0700

    Merge pull request #3853 from walkerlala/add-ade20k

    add ADE20K dataset
`

Tensorflow version -> 1.7.0
Using an AWS P3.2x instance with a standard AMI

I tried the other models and everything works just fine. RFCN and Faster RCNN with resnet/inception all work just fine. I also tried the low proposal and regular checkpoint and there doesn't seem to be a difference.
 
The error:

`WARNING:root:Variable [FirstStageFeatureExtractor/cell_8/prev_1x1/weights/Momentum] is not available in checkpoint
WARNING:root:Variable [FirstStageFeatureExtractor/cell_8/prev_bn/beta/Momentum] is not available in checkpoint
WARNING:root:Variable [FirstStageFeatureExtractor/cell_8/prev_bn/gamma/Momentum] is not available in checkpoint
WARNING:root:Variable [FirstStageFeatureExtractor/cell_9/1x1/weights/Momentum] is not available in checkpoint
WARNING:root:Variable [FirstStageFeatureExtractor/cell_9/beginning_bn/beta/Momentum] is not available in checkpoint
WARNING:root:Variable [FirstStageFeatureExtractor/cell_9/beginning_bn/gamma/Momentum] is not available in checkpoint
WARNING:root:Variable [FirstStageFeatureExtractor/cell_9/comb_iter_0/left/bn_sep_5x5_1/beta/Momentum] is not available in checkpoint
WARNING:root:Variable `

I tried reverting back to a previous commit, but it didn't see to work.... Also the checkpoints i'm using are available at -> http://download.tensorflow.org/models/object_detection/faster_rcnn_nas_coco_2018_01_28.tar.gz

Any thoughts?

**Comment by robieta** [comment]
I'm handling two similar issues (https://github.com/tensorflow/models/issues/3922, https://github.com/tensorflow/models/issues/3972), so I'm going to grab this one as well.

**Comment by robieta** [comment]
Hi. Based on https://github.com/tensorflow/models/issues/1795 it looks like those warnings may not actually be fatal. In particular the missing Momentum sort of makes sense, as if one wants to run on new data keeping the SGD momentum from the previous training would be detrimental.

Can you run some inference examples and see if you get reasonable results?

**Comment by alyahmed** [comment]
They are not fatal (i.e no crash will occur), the training will continue. However, the loss function will not decrease so essentially the network won't learn anything.

**Comment by robieta** [comment]
That feels more like a general deep learning training problem rather than a particular issue with the code. Have you done any hyperparameter optimization? Also, when you say loss are you referring to training or testing loss?

**Comment by alyahmed** [comment]
I'm referring to the training loss. it seems by the issue tracker many people are having this problem with other networks as well (SSD MobileNet V1 & V2). This seems like a code problem, my best guess is the names of the layers used in the provided checkpoints don't align with the ones used in this repository?

**Comment by robieta** [comment]
@jch1 Could you comment?

**Comment by robieta** [comment]
I'm going to close this as it appears to be the same issue as https://github.com/tensorflow/models/issues/3922. (Layer order changing) Feel free to reopen after the checkpoints are updated if the problem persists.
